node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -730,32 +788,36 @@ mod tests {
             .build();
         let tx2 = TxBuilder::default()
             .hash(H256::from_low_u64_le(2))
-            .status(TxStatus::Failed)
+            .status(TxStatus::Pending)
             .build();
         let tx3 = TxBuilder::default()
             .hash(H256::from_low_u64_le(3))
+            .status(TxStatus::Failed)
+            .build();
+        let tx4 = TxBuilder::default()
+            .hash(H256::from_low_u64_le(4))
             .status(TxStatus::Succeeded(TransactionBlock {
                 block_hash: Default::default(),
                 block_number: Default::default(),
             }))
             .build();
         subject
-            .insert_new_records(&vec![tx1.clone(), tx2.clone(), tx3.clone()])
+            .insert_new_records(&vec![tx1.clone(), tx2.clone(), tx3.clone(), tx4.clone()])
             .unwrap();
-        let hashset = HashSet::from([tx1.hash, tx2.hash]);
+        let hashset = HashSet::from([tx1.hash, tx3.hash]);
```

Comment:
When I come to think of this HashSet, I may not like it. Look, what is the benefit? Even if there are duplicates which might be fought by the nature of the HashSet we might want to panic on such an event. Hmmm...
I'm just pondering here, I have the feeling that you actually implemented that for this very reason. Then ignore my youthful senility.

Reply:
It's for the same reason, In case vector seems more convenient, I'll replace it and will add an input error check.